### PR TITLE
SPARK-715: Adds instructions for building with Maven to documentation

### PR DESCRIPTION
**Pull request by andyk**
Fixes SPARK-715 (https://spark-project.atlassian.net/browse/SPARK-715)

Basically I copied a tweaked version of the instructions in this pull request #310 to a new page in the docs and added links to that page from the appropriate places.

Adds links to new instructions in:

The main Spark project README.md
The docs nav menu called "More"
The docs Overview page under the "Building" and "Where to Go from Here" sections

### CODE REVIEW

File: docs/building-with-maven.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## Spark Tests in Maven ##
+
+Tests are run by default via the scalatest-maven-plugin. With this you can do things like:
+
+skip test execution (but not compilation):
+
+    $ mvn -DskipTests -Phadoop2 clean install
+
+To runn a specific test suite:
```
Comment (mateiz) [inline docs/building-with-maven.md]: Small typo here

File: docs/building-with-maven.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    $ mvn -Phadoop2 scala:cc
+
+…should run continuous compilation (i.e. wait for changes). However, this has not been tested extensively.
+
+## Spark Tests in Maven ##
+
+Tests are run by default via the scalatest-maven-plugin. With this you can do things like:
+
+skip test execution (but not compilation):
```
Comment (mateiz) [inline docs/building-with-maven.md]: Skip might need to be capitalized

File: docs/building-with-maven.md (hunk 1 above)
Comment (andyk) [inline docs/building-with-maven.md]: Fixed.

File: docs/building-with-maven.md (hunk 2 above)
Comment (andyk) [inline docs/building-with-maven.md]: Fixed.

**Comment by mateiz** [comment]
Cool, thanks.

**Pull request merged**